workers/ohsome_quality_analyst/indicators/mapping_saturation/models.py

```diff
@@ -0,0 +1,408 @@
+    https://en.wikipedia.org/wiki/Logistic_function
+    """
+
+    name = "Sigmoid model"
+    function_formula = "f(x) = L / (1 + e^(-k * (x - x_0)))"
+
+    def __init__(self, xdata, ydata):
+        super().__init__(xdata, ydata)
+        # curve_fit: Use non-linear least squares to fit a function, f, to data.
+        # popt: Optimal values for the parameters as array
+        # pcov: The estimated covariance of popt
+        popt, pcov = curve_fit(
+            self.function,
+            xdata=xdata,
+            ydata=ydata,
+            p0=self.initial_guess(),
+            bounds=self.bounds(),
+        )
+
+        self.coefficients = {
+            "x_0": popt[0],
+            "k": popt[1],
+            "L": popt[2],
+        }
+        self.asym_conf_int = self.confint(popt, pcov)
+        self.fitted_values = self.function(xdata, **self.coefficients)
+
+    def function(
+        self, x: np.float64, x_0: np.float64, k: np.float64, L: np.float64
+    ) -> np.float64:
+        """Sigmoid function."""
+        return L / (1 + np.exp(-k * (x - x_0)))
+
+    def initial_guess(self) -> tuple:
+        """Get an initial guess on parameters for single Sigmoid function."""
+        x_0 = self.xdata.size / 2
+        k = 0
+        L = self.ydata.max()
+        return (x_0, k, L)
+
+    def bounds(self) -> tuple:
+        """Get lower and upper bounds of the parameters for single Sigmoid function."""
+        x_0_upper_bound = self.xdata.size * 1.5
+        x_0_lower_bound = 0.0
+        k_upper_bound = 1.0
+        k_lower_bound = -1.0
+        L_upper_bound = self.ydata.max()
+        L_lower_bound = 0.0
+        return (
+            (x_0_lower_bound, k_lower_bound, L_lower_bound),
+            (x_0_upper_bound, k_upper_bound, L_upper_bound),
+        )
+
+    @property
+    def asymptote(self):
+        return self.coefficients["L"]
+
+    @property
+    def inflection_point(self):
+        return self.coefficients["x_0"]
+
+    def confint(
+        self, popt: np.array, pcov: np.array, par_pos: int = 2, level: float = 0.95
+    ) -> np.array:
+        """Confidence Intervals for a certain Model Parameter
+
+        Args:
+            pcov: The estimated covariance of the optimal values for the parameters
+            par_pos: Parameter position (index)
+            level: Confidence level
+
+        Returns:
+            np.array: lower (first element) and upper (second element) confidence limits
+                of the model parameter
+        """
+        alpha = 1.0 - level
+        degrees_of_freedom = len(self.ydata) - 3  # 3 is the number of parameters
+        tval = t_distribution.ppf(1.0 - alpha / 2.0, degrees_of_freedom)
+        perr = np.sqrt(np.diag(pcov))  # standard deviation errors
+        lower = popt - perr * tval
+        upper = popt + perr * tval
+        return np.array([lower[par_pos], upper[par_pos]])
+
+
+class SSlogis(BaseStatModel):
```

Comment:
also here you could double check if this result is realistic for Touggourt (Feature ID 4 in our regions on the website).

Again I would expect the asymptote to be higher.

![image](https://user-images.githubusercontent.com/7045979/146796723-810731dd-bb7b-4edc-8afe-87b704a5ca93.png)

Reply:
That is an interesting case. Here are all the models which produced fit.

![Figure_2](https://user-images.githubusercontent.com/83658582/146903311-138a9f10-7d87-4c81-beb9-53151e25a5ea.png)

My guess is that because of the step wise nature of the graph the curve fitting is difficult for the models. Since initial guess and bounds for the curve fitting is done by the R model itself there is not much which can be done about that. Except maybe to introduce weights to make the last data-points more important.

Another interesting edge case is that one would expect to rule this model out since the data is above the asymptote (more then the 5% margin). But the actual criterion here is  "if the latest data (average last two years) is lower than the upper limit of the 95% confidence interval of the estimated asymptote"

Reply:
Somehow the graphs I posted are not showing the same data as the graph you posted. But I am pretty sure that my graphs are showing the result for the Touggourt region. Let me do some more checks.

Reply:
@Hagellach37, your screenshot is based on an old commit where the asymptote has not been calculated correctly. Here is the current version (the percentage number format will be fixed). 
![screenshort](https://user-images.githubusercontent.com/83658582/146908773-a6eec8df-5a61-469e-b7be-b31586821896.png)

Reply:
I can't reproduce your figure. for me the completeness is 113%

![image](https://user-images.githubusercontent.com/7045979/146924638-f565822b-6ca1-4c7e-8359-8378fc5a7250.png)

Reply:
I'm a bit confused why your plot has a maximum value of around 200 for the OSM data and my plot shows a value higher than 350. Are we using the same region?